lib/linguist/heuristics.yml

```
@@ -138,6 +138,12 @@ disambiguations:
     pattern: '\/\* |\/\/ |^\}'
 - extensions: ['.cls']
   rules:
+  - language: Visual Basic 6.0
+    and:
+    - named_pattern: vb-class
+    - pattern: '^\s*BEGIN(\r\n|\n)\s*MultiUse\s*=.*(\r\n|\n)\s*Persistable\s*='
```

Using `\R` is more idiomatic and concise:

```suggestion
    - pattern: '^\s*BEGIN\R\s*MultiUse\s*=.*\R\s*Persistable\s*='
```

Also, I noticed the pattern only matches `BEGIN\R`, without accounting for trailing whitespace (`BEGIN\s*\R`). Was this intentional? (I'm not at all familiar with these languages).

---

Thanks for the review. I'm less familiar with Ruby-style regex, so that `\R` trick is good to know!

Regarding the absence of trailing spaces after `BEGIN`, this is indeed intentional because I'm relying on the VB6 IDE which doesn't add trailing spaces. Since this part of the text file is basically metadata generated by the IDE, there's no reason for developers to alter these lines.

---

> Using \R is more idiomatic and concise:

... but it also breaks matches that use other regex engines, like RE2 😢

---

I could always revert to what I had originally (`(\r\n|\n)`), but maybe it's better to take this opportunity to a find a good way to convert `\R` for other engines in case this happens again in the future.

From what I can see in the [documentation](https://ruby-doc.org/3.2.1/Regexp.html#:~:text=/%5CR/%20%2D%20A%20linebreak%3A%20%5Cn%2C%20%5Cv%2C%20%5Cf%2C%20%5Cr%20%5Cu0085%20(NEXT%20LINE)%2C%20%5Cu2028%20(LINE%20SEPARATOR)%2C%20%5Cu2029%20(PARAGRAPH%20SEPARATOR)%20or%20%5Cr%5Cn.), the long way to express `\R` would be something like: `(\n|\v|\f|\r|\u0085|\u2028|\u2029|\r\n)`.

@bzz Would that longer expression work with RE2?

---

@bzz I fail to see how RE2 is relevant here, as these heuristics are only consumed by Linguist (a Ruby program, which uses [Oniguruma](https://github.com/kkos/oniguruma/blob/master/doc/RE)-flavoured regexes).

@DecimalTurn Please stick with `\R`; it's already in-use elsewhere in `heuristics.yml` ([exempli gratia](https://github.com/DecimalTurn/linguist/blob/d790d3f4311ac80aabbcfd7b178793a1eba76da2/lib/linguist/heuristics.yml#L661)), and its inclusion isn't going to break anything.

---

@Alhadis - Compatibilty with RE2 is relevant insofar as we care about Code Search to be in line with Linguist. As described in this [blog post](https://github.blog/2021-12-15-a-brief-history-of-code-search-at-github/), GitHub now uses [Enry ](https://github.com/go-enry/go-enry)(which itself uses RE2) for language detection in Code Search. 

Divergences between Linguist and Enry can lead to some weird situations. For example, Linguist might say that a language is present in a repo when looking at the language statistics bar, but Code Search could then find no files using that language.

In the case of `\R`, I don't think it is really problematic since Enry can likely solve the issue with a simple substitution (as I mentioned above). Nonetheless, I do believe that it's a valid point of discussion to raise here. Ideally, we want to keep [the list of divergences](https://github.com/go-enry/go-enry#divergences-from-linguist) as small as possible.

---

@DecimalTurn Ah. I wasn't aware that the new Code Search was being powered by Go. That changes things.

Regarding `\R`, I'm actually wondering if it wouldn't make more sense to have Linguist replace `\n` with `(?:\r?\n|\r)` when loading `heuristics.yml`: it's unlikely that a meaningful heuristic will ever depend on one's choice of line terminator. If I'm wrong, then `\\x0A` or `\\x0D\\x0A` could be used as a workaround to circumvent the "auto-newline" feature.